denonavr/ssdp.py

```diff
@@ -174,28 +178,44 @@ def evaluate_scpd_xml(url):
             # Look for manufacturer "Denon" in response.
             # Using "try" in case tags are not available in XML
             device = {}
+            device_xml = None
             device["manufacturer"] = (
                 root.find(SCPD_DEVICE).find(SCPD_MANUFACTURER).text)
 
             _LOGGER.debug("Device %s has manufacturer %s", url,
                           device["manufacturer"])
-            if (device["manufacturer"] in SUPPORTED_MANUFACTURERS and
-                    root.find(SCPD_DEVICE).find(
-                        SCPD_DEVICETYPE).text == DEVICETYPE_DENON):
+
+            if not device["manufacturer"] in SUPPORTED_MANUFACTURERS:
+                return False
+
+            if (root.find(SCPD_DEVICE).find(SCPD_DEVICETYPE).text
+                    in SUPPORTED_DEVICETYPES):
+                device_xml = root.find(SCPD_DEVICE)
+            elif root.find(SCPD_DEVICE).find(SCPD_DEVICELIST):
+                for dev in root.find(SCPD_DEVICE).find(SCPD_DEVICELIST):
+                    if (dev.find(SCPD_DEVICETYPE).text in SUPPORTED_DEVICETYPES
+                            and dev.find(SCPD_SERIALNUMBER) is not None):
+                        device_xml = dev
+                        break
+
+            if not device_xml:
```

Comment:
This would create the same FutureWarning as solved here:
https://github.com/scarface-4711/denonavr/pull/129/commits/cd168d01d2725806a671852378c8bfe47d12795d#diff-18d32fba6089c732c810ffa526db114cR1031

Reply:
Done